Py time (#669)
Codecademy/docs: content/python/concepts/time-module/time-module.md

---
Title: 'Time Module'

> Description: 'Python has a built-in time module that we can use for assorted time functions. The following functions are provided by this module.'

Reviewer: ```suggestion
Description: 'Python has a built-in time module that we can uses functions for working with time.'
```
```suggestion
Description: 'Python has a built-in time module that we can use for assorted time functions. The following functions are provided by this module.'
```

Subjects:
  - 'Computer Science'
  - 'Data Science'
  - 'Web Development'
Tags:
  - 'Methods'
  - 'Constants'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
  - 'paths/front-end-engineer-career-path'
---

Python has a built-in `time` module that we can use for time related tasks.

> The following functions are provided by this module.

Reviewer: I think  at this line, we should include some brief information about terms, abbreviations, and phrases that are commonly used when working with the `time` module, such as:

* The epoch and how it is platform-dependent.
  * How seconds and leap-seconds are calculated according to the epoch.
* Timezones like UTC (formerly GMT).
* Certain events like daylights savings time (though I wonder how things might be effected if it's done away in the US if this [proposed bill](https://www.reuters.com/world/us/us-senate-approves-bill-that-would-make-daylight-savings-time-permanent-2023-2022-03-15/) is signed into law).

Maybe we should make a separate `struct_time` entry for a future PR.

Author: I think a struct time entry is worthwhile

> The following functions are provided by this module.

Reviewer: I think  at this line, we should include some brief information about terms, abbreviations, and phrases that are commonly used when working with the `time` module, such as:

* Epoch and how it is platform-dependent.
  * How seconds and leap-seconds are calculated according to the epoch.
* Timezones like UTC (formerly GMT).
* Certain events like daylights savings time (though I wonder how things might be effected if it's done away in the US if this [proposed bill](https://www.reuters.com/world/us/us-senate-approves-bill-that-would-make-daylight-savings-time-permanent-2023-2022-03-15/) is signed into law).

Maybe we should make a separate `struct_time` entry for a future PR.

---
Title: 'Time Module'

> Description: 'Python has a built-in time module that has functions we can use for working with time.'

Reviewer 2: ```suggestion
Description: 'Python has a built-in time module that has functions that can be used for working with time.'
```

Subjects:
  - 'Computer Science'
  - 'Data Science'
  - 'Web Development'
Tags:
  - 'Methods'
  - 'Constants'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
  - 'paths/front-end-engineer-career-path'
---

> Python has a built-in `time` module that we can use for time related tasks. This module utilizes a number of different standards and objects that may be unfamiliar such as:

Reviewer 2: ```suggestion
Python has a built-in `time` module that that can be used for time related tasks. This module utilizes a number of different standards and objects that may be unfamiliar such as:
```

- The epoch refers to a reference point in time from which calculations can be made. It is designated as midnight on January 1, 1970.

> - There are functions that will return values formatted using Daylight Savings Time (DST).

Reviewer 2: I don't think using Daylight Savings Time would be that unfamiliar.

Author: It probably is relatively familiar, but it's also not a universal convention and many countries do not adjust the hour as we do. Additionally, it is a standard that is potentially in flux, there is currently a movement in the US to move permanently to DST.
